modules/apps/bulk/bulk-rest-test/src/testIntegration/java/com/liferay/bulk/rest/resource/v1_0/test/BulkActionResourceTest.java

```diff
@@ -781,6 +783,153 @@ private void _testPostBulkActionWithTypeDefaultPermission()
+
+		ObjectDefinition objectDefinition =
+			ObjectDefinitionLocalServiceUtil.
+				getObjectDefinitionByExternalReferenceCode(
+					"L_CMS_DEFAULT_PERMISSION", _depotEntry2.getCompanyId());
+
+		ObjectEntryFolder objectEntryFolder1 =
+			_objectEntryFolderLocalService.
+				getObjectEntryFolderByExternalReferenceCode(
+					ObjectEntryFolderConstants.EXTERNAL_REFERENCE_CODE_CONTENTS,
+					_depotEntry2.getGroupId(), _depotEntry2.getCompanyId());
+
+		ObjectEntryFolder objectEntryFolder2 =
+			_objectEntryFolderLocalService.addObjectEntryFolder(
+				RandomTestUtil.randomString(), _depotEntry2.getGroupId(),
+				TestPropsValues.getUserId(),
+				objectEntryFolder1.getObjectEntryFolderId(), "",
+				HashMapBuilder.put(
+					LocaleUtil.ENGLISH, RandomTestUtil.randomString()
+				).build(),
+				RandomTestUtil.randomString(),
+				ServiceContextTestUtil.getServiceContext());
+
+		ObjectEntryFolder objectEntryFolder3 =
+			_objectEntryFolderLocalService.addObjectEntryFolder(
+				RandomTestUtil.randomString(), _depotEntry2.getGroupId(),
+				TestPropsValues.getUserId(),
+				objectEntryFolder1.getObjectEntryFolderId(), "",
+				HashMapBuilder.put(
+					LocaleUtil.ENGLISH, RandomTestUtil.randomString()
+				).build(),
+				RandomTestUtil.randomString(),
+				ServiceContextTestUtil.getServiceContext());
+
+		JSONObject defaultPermissionsJSONObject = JSONUtil.put(
+			ObjectEntryFolderConstants.EXTERNAL_REFERENCE_CODE_CONTENTS,
+			JSONUtil.put(
+				role1.getName(), JSONUtil.putAll(ActionKeys.VIEW)
+			).put(
+				role2.getName(), JSONUtil.putAll(ActionKeys.VIEW)
+			));
+
+		defaultPermissionBulkAction.setDefaultPermissions(
+			defaultPermissionsJSONObject.toString());
+
+		defaultPermissionBulkAction.setDepotGroupId(_depotEntry2.getGroupId());
+		defaultPermissionBulkAction.setBulkActionItems(
+			new BulkActionItem[] {
+				_toBulkActionItem(objectEntryFolder2),
+				_toBulkActionItem(objectEntryFolder3)
+			});
+
+		defaultPermissionBulkAction.setType(
+			BulkAction.Type.DEFAULT_PERMISSION_BULK_ACTION);
+
+		_postBulkAction(defaultPermissionBulkAction);
+
+		JSONObject jsonObject = _getDefaultPermissionsJSONObject(
+			objectDefinition, objectEntryFolder2);
+
+		defaultPermissionsJSONObject = jsonObject.getJSONObject(
+			ObjectEntryFolderConstants.EXTERNAL_REFERENCE_CODE_CONTENTS);
+
+		JSONArray jsonArray = defaultPermissionsJSONObject.getJSONArray(
+			role1.getName());
+
+		String defaultPermissions = jsonArray.toString();
+
+		Assert.assertTrue(defaultPermissions.contains(ActionKeys.VIEW));
+
+		jsonArray = defaultPermissionsJSONObject.getJSONArray(role2.getName());
+
+		defaultPermissions = jsonArray.toString();
+
+		Assert.assertTrue(defaultPermissions.contains(ActionKeys.VIEW));
+
+		JSONObject singleRoleDefaultPermissionsJSONObject = JSONUtil.put(
+			ObjectEntryFolderConstants.EXTERNAL_REFERENCE_CODE_CONTENTS,
+			JSONUtil.put(
+				role1.getName(),
+				JSONUtil.putAll(ActionKeys.DELETE, ActionKeys.UPDATE)));
```

Comment:
I think this is wrong. I mean, in reality we are posting the FULL list of roles but we then use just one. Otherwise in this case we don't know if we don't update the other roles, they are not in the json.

I think this should be like the defaultPermissionsJSONObject variabile (I think you can use the same just putting again the role1 with different permissions